src/Symfony/Component/Process/ExecutableFinder.php

```diff
@@ -75,10 +66,17 @@ public function find(string $name, string $default = null, array $extraDirs = []
             foreach ($dirs as $dir) {
                 if (@is_file($file = $dir.\DIRECTORY_SEPARATOR.$name.$suffix) && ('\\' === \DIRECTORY_SEPARATOR || @is_executable($file))) {
                     return $file;
+                } elseif(basename($dir) === $name.$suffix && @is_executable($dir)) {
+                    return $dir;
                 }
             }
         }
 
+        $command = '\\' === \DIRECTORY_SEPARATOR ? 'where' : 'command -v';
+        if (\function_exists('exec') && ($executablePath = strtok(exec($command.' '.escapeshellarg($name)), \PHP_EOL)) && is_executable($executablePath)) {
```

Comment:
shouldn't we add the `function_exists('exec')` check to PhpExecutableFinder also?

Reply:
Fixed.